### PR TITLE
Fix regex to allow wildcard subdomain

### PR DESCRIPTION
**Pull request by SvyatovetsD**
I couldn't add hostname binding with wildcard subdomain though it can be done via azure portal. Tweaked regex expression a bit, so it could match wildcard

### CODE REVIEW

File: src/ResourceManagement/AppService/HostNameBindingImpl.cs [hunk 1]
```diff
@@ -140,7 +140,7 @@ private string NormalizeHostNameBindingName(string hostname, string domainName)
         ///GENMHASH:027705B70337BE533ED77421800E496A:3479885A232C974264B5B36BDBB0BC94
         public HostNameBindingImpl<FluentT, FluentImplT, DefAfterRegionT, DefAfterGroupT, UpdateT> WithDnsRecordType(CustomHostNameDnsRecordType hostNameDnsRecordType)
         {
-            var regex = new Regex("([.\\w-]+)\\.([\\w-]+\\.\\w+)");
+            var regex = new Regex("([.\\w-]+|[*])\\.([\\w-]+\\.\\w+)");
```
Comment (ChenTanyi) [inline src/ResourceManagement/AppService/HostNameBindingImpl.cs:143]: It is better not to add `[` and `]`.

**Comment by weidongxu-microsoft** [comment]
This seems not allowed on portal.

image

**Review by ChenTanyi — COMMENTED** [review]
Hi @SvyatovetsD, we cannot use bind a web app with wildcard subdomain in Azure Portal, would you like to show a screenshot, gif or docs for it?

File: src/ResourceManagement/AppService/HostNameBindingImpl.cs (hunk 1 above)
Comment (SvyatovetsD) [reply src/ResourceManagement/AppService/HostNameBindingImpl.cs:143]: It seems to work fine in my environment. I guess you got an error cause dns zone haven't been setup accordingly
Screenshot 2020-11-18 141755

Comment (weidongxu-microsoft) [reply src/ResourceManagement/AppService/HostNameBindingImpl.cs:143]: @SvyatovetsD
Could you see if you can apply @ChenTanyi 's advice above?

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**